### PR TITLE
Release/1.0.0

### PR DESCRIPTION
**Pull request by ThinkTankShark**

### CODE REVIEW

File: src/Button.jsx [hunk 1]
```diff
@@ -48,21 +48,12 @@ export default class Button extends Component {
...
-    } else if (this.props.disabled !== nextProps.disabled) {
-      return true;
-    } else if (this.state.hoverSize !== nextState.hoverSize) {
-      return true;
-    } else if (this.props.iconClass !== nextProps.iconClass) {
-      return true;
-    } else {
-      return false;
-    }
+    const propsToCheck = ['children', 'className', 'iconClass', 'disabled'];
```
Comment (isuttell) [inline src/Button.jsx]: We can't compare children strictly since we don't know if it's going to be immutable

File: src/Button.jsx [hunk 2]
```diff
@@ -83,9 +75,20 @@ export default class Button extends Component {
...
+    const NextHoverSize = Math.max(
+      this.refs.container.offsetWidth,
+      this.refs.container.offsetHeight
+    );
+    const currHoverSize = this.state.hoverSize;
+    const hoverSizeChanged = NextHoverSize !== currHoverSize;
+
+    // Only update the hoverSize state if
+    // it's not the same as the current state.hoverSize
+    if (hoverSizeChanged && this.refs.container) {
```
Comment (isuttell) [inline src/Button.jsx]: If `this.refs.container` is not defined it's going to break on line 79. We should be checking it there

File: src/Button.jsx [hunk 3]
```diff
@@ -48,21 +48,13 @@ export default class Button extends Component {
...
-      return true;
-    } else if (this.props.iconClass !== nextProps.iconClass) {
-      return true;
-    } else {
-      return false;
-    }
+    const propsToCheck = ['className', 'iconClass', 'disabled'];
+    const statesToCheck = ['hover', 'pressed', 'hoverSize'];
+    return (
+      this.props.forceUpdate ||
```
Comment (isuttell) [inline src/Button.jsx]: This should probably be `nextProps.forceUpdate` incase there is a time that it changes

File: README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
...
+[![npm](https://img.shields.io/npm/v/ship-components-buttons.svg)](https://www.npmjs.com/package/ship-components-buttons) [![coverage](https://img.shields.io/coveralls/ship-components/ship-components-buttons.svg)](https://coveralls.io/github/ship-components/ship-components-buttons) [![Build Status](http://img.shields.io/travis/ship-components/ship-components-buttons/master.svg?style=flat)](https://travis-ci.org/ship-components/ship-components-buttons) [![dependencies](https://img.shields.io/david/ship-components/ship-components-buttons.svg?style=flat)](https://david-dm.org/ship-components/ship-components-buttons) [![devDependencies](https://img.shields.io/david/dev/ship-components/ship-components-buttons.svg?style=flat)](https://david-dm.org/ship-components/ship-components-buttons?type=dev)
```
Comment (isuttell) [inline README.md]: What is actually changing here?

File: src/Button.jsx [hunk 5]
```diff
@@ -83,9 +75,20 @@ export default class Button extends Component {
    * Get the width of the button so so we ensure the hover effect fits
    */
   calculateWidth() {
-    this.setState({
-      hoverSize: this.refs.container ? Math.max(this.refs.container.offsetWidth, this.refs.container.offsetHeight) : void 0
-    });
+    const NextHoverSize = Math.max(
```
Comment (isuttell) [inline src/Button.jsx]: Stylistically this should be `nextHoverSize`. Only classes or other functions we call `new` on are capitalized.

File: README.md (hunk 4 above)
Comment (ThinkTankShark) [reply README.md]: I removed the ?maxAge=259000 from 2 links to make sure it updates right away

File: src/Button.jsx (hunk 5 above)
Comment (ThinkTankShark) [reply src/Button.jsx]: Fixed, typo

File: src/Button.jsx (hunk 3 above)
Comment (ThinkTankShark) [reply src/Button.jsx]: Right, fixed

File: src/Button.jsx (hunk 2 above)
Comment (ThinkTankShark) [reply src/Button.jsx]: Right, fixed

**Pull request merged**